### ISSUE TITLE
Weird output for net.predict() and net.get_output() 

### ISSUE THREAD

**Issue by gunsl1nger** [dnouri/nolearn#212]
Hello, this interface is really cooool to use, I'm using it for an autoencoder. I just noticed though that when I train a 256 - 100 - 256 (example) autoencoder, and then use predict(X) or get_output(X) on the middle layer I get the same values on output no matter if X is normalized or standardized or just left as is. I got over this problem by just multiplying with the weight matrix and adding the bias but now when the autoencoder is unwinded (8-9 layers) that's kinda expensive. 

I hope I'm wrong and have a mistake somewhere :)

**Comment by BenjaminBossan** [comment]
Hi, I think a minimal working example could be helpful here.

**Comment by gunsl1nger** [comment]
def model_for_layerwise_training(input_units=0, hidden_units=0, output_units=0):
    """"
    Layer-wise training for the autoencoder.
    :param input_units: Number of neurons in the input layer.
    :param hidden_units: Number of neurons in the hidden layer.
    :param output_units: Number of neurons in the output layer.
    :return: A neural network model.
    """""

    net = NeuralNet(
        layers=[('input', layers.InputLayer),
                ('hidden', layers.DenseLayer),
                ('output', layers.DenseLayer),
                ],

        # Layer parameters
        input_shape=(None, input_units),
        hidden_num_units=hidden_units,
        output_num_units=output_units,

        hidden_nonlinearity=lasagne.nonlinearities.sigmoid,
        output_nonlinearity=None,

        # Optimization. Learning rate and momentum are variable of time.
        update=nesterov_momentum,
        update_learning_rate=theano.shared(float32(0.03)),
        update_momentum=theano.shared(float32(0.9)),

        objective_loss_function=lasagne.objectives.squared_error,

        regression=True,
        on_epoch_finished=[
            AdjustVariable('update_learning_rate', start=0.03, stop=0.0001),
            AdjustVariable('update_momentum', start=0.9, stop=0.999),
            ],
        max_epochs=500,
        verbose=1
    )

    return net

model = model_for_layerwise_training(256, 100, 256)
features = np.load('features.npy')
model.fit(features, features)
model.save_params_to('parameters')
middle_layer = layers.get_output(model.layers_['hidden'], deterministic=True)
model_predicted = model.get_output(middle_layer, X)) 
parameters = open('parameters')
params = pickle.load(parameters)
hidden_matrix = params['hidden'][0]
hidden_bias = params['hidden'][1]
predicted_by_multiplication = X.dot(hidden_matrix) + hidden_bias
print (predicted_by_multiplication - model_predicted)

Here is something, but I cannot test the example out because I have some problems with the update, nothing compiles.
